src/routes/user.ts

```diff
@@ -0,0 +1,88 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { Type, Static } from '@sinclair/typebox';
+import { knexController } from '../database/database';
+
+const userAddedResponse = Type.Object({
+  status: Type.String(),
+});
+
+export type UserAddedResponse = Static<typeof userAddedResponse>;
+
+const schema = {
+  description: 'Add new user or return error if add is not successful',
+  summary: 'Add user',
+  tags: ['User'],
+  body: {
+    type: 'object',
+    required: ['email', 'forename', 'surname', 'password'],
+    properties: {
+      email: {
+        type: 'string',
+      },
+      forename: {
+        type: 'string',
+      },
+      surname: {
+        type: 'string',
+      },
+      password: {
+        type: 'string',
+      },
+    },
+  },
+  response: {
+    201: {
+      description: 'User added',
+      type: 'null',
+    },
+    409: {
+      description: 'Email already in use',
+      type: 'null',
+    },
+    500: {
+      description: 'Server error',
+      type: 'null',
+    },
+  },
+};
+
+const handler = async (
+  req: FastifyRequest,
+  reply: FastifyReply
+): Promise<void> => {
+  if (req.body) {
+    // it has it believe or not...
+    const result = await knexController('user')
+      .count()
+      .where('email', req.body['email']);
+    const count: Number = Number(result[0]['count(*)']);
+    if (count > 0) {
+      reply.code(409);
+      await reply.send();
```

Comment:
```suggestion
      return reply.code(409).send({statusCode: 409, error: 'Conflict', message: 'Jottain men nyt pieleen, tais olla sähköposti käytössä hehe'});
```

Reply:
Jos alat oikein velhoksi, niin rykäse meille samalla funktio joka palauttaa tommoisen error objektin. Esim näin `errorMessage(statusCode: number, message: string | undefined)` niin se etsisi automaagisesti ton errorin (conflict) ja lisäisi sen mukaan.

Tähän voinee löytyä joku kirjasto

Reply:
Mutta ei oo kovin hankala itsekkään hallinnoida jotain yleisimpien koodien erroria, esim 
```
const errors = {400: 'Bad Request', 409: 'Conflict', 500: 'Internal Server Error'}
```

Reply:
Asia saattaa muuttua riippuen siitä miten toteutamme käyttäjän disabloimisen/deletoimisen. Siitä keskustellaan sitten